packages/ra-core/src/form/FormWithRedirect.tsx

```diff
@@ -115,9 +132,17 @@ const FormWithRedirect = ({
             },
             registerField: (source, group) => {
                 if (group) {
-                    const fields = new Set(formGroups.current[group] || []);
-                    fields.add(source);
-                    formGroups.current[group] = Array.from(fields);
+                    if (!(formGroups.current[group] || []).includes(source)) {
```

Comment:
you've added more code but not more tests, I'm a bit scared

Reply:
Because the behavior has not changed, hence the tests related to form groups still pass (`TabbedForm` and `TranslatableInput`)